python/descriptor_pool.c

```diff
@@ -0,0 +1,216 @@
+
+  const char* name = PyUpb_GetStrData(arg);
+  if (!name) {
+    return NULL;
+  }
+
+  const upb_fielddef* field = upb_symtab_lookupext(self->symtab, name);
+  if (field == NULL) {
+    return PyErr_Format(PyExc_KeyError, "Couldn't find extension %.200s", name);
+  }
+
+  return PyUpb_FieldDescriptor_GetOrCreateWrapper(field, _self);
+}
+
+static PyMethodDef PyUpb_DescriptorPool_Methods[] = {
+    /*
+      TODO: implement remaining methods.
+    { "Add", Add, METH_O,
+      "Adds the FileDescriptorProto and its types to this pool." },
+      */
+    {"AddSerializedFile", PyUpb_DescriptorPool_AddSerializedFile, METH_O,
+     "Adds a serialized FileDescriptorProto to this pool."},
+    /*
+    { "FindFileByName", FindFileByName, METH_O,
+      "Searches for a file descriptor by its .proto name." },
+    { "FindMessageTypeByName", FindMessageByName, METH_O,
+      "Searches for a message descriptor by full name." },
+    { "FindFieldByName", FindFieldByNameMethod, METH_O,
+      "Searches for a field descriptor by full name." },
+      */
+    {"FindExtensionByName", PyUpb_DescriptorPool_FindExtensionByName, METH_O,
+     "Searches for extension descriptor by full name."},
+    /*
+    { "FindEnumTypeByName", FindEnumTypeByNameMethod, METH_O,
+      "Searches for enum type descriptor by full name." },
+    { "FindOneofByName", FindOneofByNameMethod, METH_O,
+      "Searches for oneof descriptor by full name." },
+    { "FindServiceByName", FindServiceByName, METH_O,
+      "Searches for service descriptor by full name." },
+    { "FindMethodByName", FindMethodByName, METH_O,
+      "Searches for method descriptor by full name." },
+    { "FindFileContainingSymbol", FindFileContainingSymbol, METH_O,
+      "Gets the FileDescriptor containing the specified symbol." },
+    { "FindExtensionByNumber", FindExtensionByNumber, METH_VARARGS,
+      "Gets the extension descriptor for the given number." },
+    { "FindAllExtensions", FindAllExtensions, METH_O,
+      "Gets all known extensions of the given message descriptor." },
+    */
+    {NULL}};
+
+static PyType_Slot PyUpb_DescriptorPool_Slots[] = {
+    {Py_tp_new, PyUpb_DescriptorPool_New},
+    {Py_tp_dealloc, PyUpb_DescriptorPool_Dealloc},
+    {Py_tp_methods, PyUpb_DescriptorPool_Methods},
+    {0, NULL}};
+
+static PyType_Spec PyUpb_DescriptorPool_Spec = {
+    PYUPB_MODULE_NAME ".DescriptorPool",  // tp_name
+    sizeof(PyUpb_DescriptorPool),         // tp_basicsize
+    0,                                    // tp_itemsize
+    Py_TPFLAGS_DEFAULT,                   // tp_flags
+    PyUpb_DescriptorPool_Slots,
+};
+
+// -----------------------------------------------------------------------------
+// Top Level
+// -----------------------------------------------------------------------------
+
+bool PyUpb_InitDescriptorPool(PyObject* m) {
+  PyTypeObject* descriptor_pool_type =
+      AddObject(m, "DescriptorPool", &PyUpb_DescriptorPool_Spec);
+
+  if (!descriptor_pool_type) return false;
```

Comment:
you are inconsistent within this file if you do the 1 line or 3 line if for an early return like this.  I prefer 1, but don't really care as long as you are consistent

Reply:
Done.